spec.md

```diff
@@ -1492,6 +1502,187 @@ Applying integer decoding with a divisor of `100` to create an array of 32-bit f
+
+The following are fields that are __not__ supplied by the *RCSB*, and are provided for applications to transfer non-molecular data along with the molecular data.
+
+Each field is a [map](#types), and there are 6 possible fields `bondProperties`, `atomProperties`, `groupProperties`, `chainProperties`,
+`modelProperties`, and `extraProperties`.  The first 5 fields are restricted to holding data that is relevant to the field's prefix, while
+any other data can be stored in the `extraProperties` field.  
+
+As mentioned in [Types](#types) in some fields we restrict the key-value types to specific types. The current type format the sepc follows is:
+* <String, Array|Binary>
+  * `atomProperties`
+  * `bondProperties`
+  * `groupProperties`
+  * `chainProperties`
+  * `modelProperties`
+* <String, String|Float|Integer|Map|Array|Binary>
+  * `extraProperties`
+
+For each of the fields above, we do reserve some key-value pair formats to help ensure cross-application-compatibility.  If you feel
+a certain key-value pair should be reserved and made a part of the spec, please let us know!  
+
+We encourage you to apply our encoding techniques to your application data to reduce file sizes!
+
+The restrictions on each of the lengths of the values in these fields are __not__ enforced at decode-time but are reserved to help guide applications.
+
+| Name            | length-restrictions     | 
+|-----------------|-------------------------|
+| bondProperties  | length of numBonds      |
+| atomProperties  | length of numAtoms      |
+| groupProperties | length of numGroups     |
+| chainProperties | length of numChains     |
+| modelProperties | length of numModels     |
+| extraProperties | None                    |
+
+Example
+
+```
+data = {
+  "mmtfVersion": "1.1",
+  "numAtoms": 999,
+  "numModels": 2,
+  "numChains": 4,
+  ...
+  "xCoordList": [1.2, 3.4, ...],
+  "yCoordList": [5.6, 7.8, ...],
+  "zCoordList": [9.0, 1.2, ...],
+  ...
+  "structureProperties": {
+    "foo_id": "ABC",
+  },
+  "modelProperties": {
+    # lists have len numModels=2
+    "foo_rmsdList": [0.5, 0.8],
+    "foo_scoreList": [1.2, 3.4],
+  },
+  "chainProperties": {
+    # lists have len numChains=4
+    "foo_uniprotIdList": ["HBB_HUMAN", "HBA_HUMAN", "HBB_HUMAN", "HBA_HUMAN"],
+    "foo_chainColorList": [0xFF0000, 0x00FF00, 0xFF0000, 0x00FF00],
+  },
+  "groupProperties": {
+    # lists have len numGroups
+    "stride_secStructList": [7, 7, 7, ...],
+    "sst_secStructList": [7, 7, 7, ...],
+  },
+  "atomProperties": {
+    # lists have len numAtoms=999
+    "pymol_colorList": [1, 2, 3, ...],
+    "pymol_repsList": [1, 1, 1, ...],
+    "apbs_chargeList": [0.1, -0.4, 0.7, ...],
+    "apbs_radiusList": [1.2, 1.8, 1.5, ...],
+  },
+  "bondProperties": {
+    # lists have len numBonds
+    "pymol_bondTypeList": [1, 1, 1, 4, 4, 4, 4, 4, 4, 1, ...],
+  },
+  "extraProperties": {
+    "pymol_bondTypes": {0: "metal", 1: "single", 2: "double", 3: "triple", 4: "aromatic"}
+  }
```

Comment:
One note on type restrictions for `extraProperties`: the example here doesn't fit since the key is an `Integer` and not a `String`. So we should either drop any second level type restriction on `extraProperties` or change the example. I vote for the former...

Reply:
I think the spec of `Map<String, String|Float|Integer|Map|Array|Binary>` sort of implies 'anything goes' but I can add a note to mention that there are no restrictions on the second level of extraProperties

Reply:
Oops you are right. I was looking at the third level here. Please ignore my comment above.

Reply:
oh well i added an example anyway! I think it is ok to be extra verbose for now and then once we have some experience working with this we can make some changes.